src/vcon/MarketGovernance.sol

```diff
@@ -0,0 +1,699 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity =0.8.13;
+
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+import {PCVMover} from "@voltprotocol/pcv/PCVMover.sol";
+import {Constants} from "@voltprotocol/Constants.sol";
+import {CoreRefV2} from "@voltprotocol/refs/CoreRefV2.sol";
+import {IPCVOracle} from "@voltprotocol/oracle/IPCVOracle.sol";
+import {IPCVDeposit} from "@voltprotocol/pcv/IPCVDeposit.sol";
+import {IPCVDepositV2} from "@voltprotocol/pcv/IPCVDepositV2.sol";
+import {IMarketGovernance} from "@voltprotocol/vcon/IMarketGovernance.sol";
+import {DeviationWeiGranularity} from "@voltprotocol/utils/DeviationWeiGranularity.sol";
+
+import {console} from "@forge-std/console.sol";
+
+/// @notice this contract requires the PCV Controller and Locker role
+///
+/// Core formula for market governance rewards:
+///     Profit Per VCON = Profit Per VCON + ∆Cumulative Profits (Dollars) * VCON:Dollar  / VCON Staked
+///
```

Comment:
add system invariants:

totalVcon = all deposits and realized gains and losses

totalUnrealizedVcon = all unrealized VCON profits and losses for a given set of venues

lazy evaluation is used across all venues to apply gains and losses in a gas efficient manner